checker.cpp

```diff
@@ -2,21 +2,57 @@
 #include <iostream>
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
-  }
-  return true;
+class Battery{
+    public:
+        float temperature,soc,chargeRate;
+    bool checkTempIsInRange(float t);
+    bool checkSocIsInRange(float s);
+    bool checkChargeRateIsInRange(float c);
+    bool checkBatteryIsOk(float t,float s ,float c);
+};
+
+class NewBattery:public Battery
+{  
+    public:
+    bool checkTempIsInRange(float temperature)
+    {
+        if(temperature < 0 || temperature > 45)
+            return false;
+        else 
+        return true;
+    }
+    bool checkSocIsInRange(float soc)
+    {
+        if(soc < 20 || soc > 80)
+            return false;
+        else
+            return true;        
+    }
+    bool checkChargeRateIsInRange(float chargeRate)
+    {
+        if(chargeRate>0.8)
+            return false;
+        else 
+            return true;
+    }
+    bool checkBatteryIsOk(float t,float s ,float c)
+    {
+        return(checkTempIsInRange(t)&&checkSocIsInRange(s)&&checkChargeRateIsInRange(c));
+    }
+};
+
+void checkBattery()
+{
+    NewBattery n;
+    assert(n.checkBatteryIsOk(0,19,1)==false);
+    assert(n.checkBatteryIsOk(1,21,1)==false);
+    assert(n.checkBatteryIsOk(1,81,0.7)==false); 
```

Comment:
Try to cover boundary conditions, out of range values and in range values, for example test values for temperature could be -1, 0, 25, 45, 50. Same can be applied to other parameters too.

Reply:
Boundary conditions i.e extremes are considered. For temperature: 0,1,44 are considered. For soc, 21,81, 79 are considered. For chargerate 0.7 and 1 are used.